internal/service/remotecfg/remotecfg.go

```diff
@@ -259,7 +277,12 @@ func (s *Service) Update(newConfig any) error {
 			newArgs.URL,
 		)
 	}
-	s.args = newArgs // Update the args as the last step to avoid polluting any comparisons
+	// Combine the new attributes on top of the system attributes
+	s.attrs = s.systemAttrs
+	maps.Copy(s.attrs, newArgs.Attributes)
```

Comment:
Does this unintentionally update `systemAttrs` with values from `newArgs.Attributes` for subsequent calls to `Update`?

Here's my thinking: 

`s.attrs = s.systemAttrs` makes `s.attrs` point to the same map in memory of the original `systemAttrs`. Copying the values from `newArgs.Attributes` into `s.attrs` (which is now the same pointer as `s.systemAttrs`) ultimately copies them into `s.systemAttrs`, which will then be the starting point for future calls.

So if you had the following:
```
remotecfg {
    attributes = {"foo" = "bar", "env" = "prod"}
}
```

then set it to:
```
remotecfg {
    attributes = {"foo" = "bar"} // Remove "env", it _shouldn't_ be present anymore
}
```
`"env"="prod"` would continue to be sent since `systemAttrs` contains it now.

I think this would fix it:
```suggestion
        newAttrs, _ := getSystemAttributes()
	maps.Copy(newAttrs, newArgs.Attributes)
	s.attrs = newAttrs
```

Reply:
I forgot to use maps.Clone here. I only want to call getSystemAttributes once on New so I don't have to recalculate anything now or added in the future.